core/encoding/json/parser.odin

```diff
@@ -263,13 +264,10 @@ parse_object_body :: proc(p: ^Parser, end_token: Token_Kind) -> (obj: Object, er
 			return
 		}
 
-		if len(obj) == cap(obj) {
-			reserve_error := reserve(&obj, max(1, cap(obj) * 2))
-			if reserve_error != nil {
-				return nil, .Out_Of_Memory
-			}
+		insert_success := runtime.map_insert(&obj, key, elem)
+		if insert_success == nil {
```

Comment:
Might be worth calling it `inserted_element_pointer` or something, so that the if makes more sense.
Or could just `if map_insert(...) == nil {` instead?